src/IO/S3/copyS3File.cpp

```diff
@@ -601,29 +601,10 @@ namespace
 
         void performCopy()
         {
-            if (size <= upload_settings.max_single_operation_copy_size)
```

Comment:
After `ComposeObject` had been added, it should be possible to use `CopyObject` in GCS, or maybe you have some different observations?

P.S. I've tested copy after multi-part upload, and now it works even without compose, but I guess not always, but when I tested the same thing half a year ago, it failed. so apparently something changed in GCS.

Reply:
`CopyObject` is used, this change avoids multi-part copy which is not implemented in GCS.

GCP support team told me that they are planning to roll out fixes related to Compose (503 for different part sizes) by the end of this month. Also, they mentioned support for MPU copy and Rewrite with the same deadline but we will see.

Reply:
> From the other side CopyObject may not work without ComposeObject right?
Yes, that was the original motivation.

> so files that had been written with preivous versions will not be able to be copiable.

It's easy to write a script to run `gsutil compose object object` retroactively on all objects in some bucket so you can make it copiable.

Also, I had the same observations as you, it is really random when and if at all it will work. But it seems they actually fixed it: https://cloud.google.com/storage/docs/release-notes#June_23_2023

Reply:
Since Compose is not required anymore, what do you think about disabling it by default? https://github.com/ClickHouse/ClickHouse/pull/58343